app/controllers/maintenance_tasks/tasks_controller.rb

```diff
@@ -33,11 +33,11 @@ def run
       redirect_to(task_path(task))
     rescue ActiveRecord::RecordInvalid => error
       redirect_to(task_path(error.record.task_name), alert: error.message)
-    rescue ActiveRecord::ValueTooLong => error
-      task_name = params.fetch(:id)
-      redirect_to(task_path(task_name), alert: error.message)
     rescue Runner::EnqueuingError => error
       redirect_to(task_path(error.run.task_name), alert: error.message)
+    rescue ActiveRecord::ValueTooLong, Parameters::TypeError => error
```

Comment:
Moving errors where we don't have access to the `task_name` any way other than via the original `params` to the bottom here. I think we can clean up some of this later on (ie. just use the id from the params for all of them), but not in this PR 😁

Reply:
There should probably be something similar in Runner to support the CLI as well.